src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Elasticsearch/V7/Transport_Request_Integration.cs

```diff
@@ -0,0 +1,79 @@
+        MinimumVersion = ElasticsearchV7Constants.Version7,
+        MaximumVersion = ElasticsearchV7Constants.Version7,
+        IntegrationName = ElasticsearchV7Constants.IntegrationName)]
+    // ReSharper disable once InconsistentNaming
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class Transport_Request_Integration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="THttpMethod">The type of the HttpMethod parameter</typeparam>
+        /// <typeparam name="TPostData">The type of the PostData parameter</typeparam>
+        /// <typeparam name="TRequestParameters">The type of the request parameters</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="method">The HTTP method of the request</param>
+        /// <param name="path">The path of the request</param>
+        /// <param name="postData">The payload of the request</param>
+        /// <param name="requestParameters">The parameters of the request</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, THttpMethod, TPostData, TRequestParameters>(TTarget instance, THttpMethod method, string path, TPostData postData, TRequestParameters requestParameters)
+        {
+            var scope = ElasticsearchNetCommon.CreateScope(Tracer.Instance, ElasticsearchV7Constants.IntegrationId, path, method.ToString(), requestParameters, out var tags);
+
+            return new CallTargetState(scope, tags);
+        }
+
+        /// <summary>
+        /// OnMethodEnd callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TResponse">Type of the response</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="response">Response instance</param>
+        /// <param name="exception">Exception instance in case the original code threw an exception.</param>
+        /// <param name="state">Calltarget state value</param>
+        /// <returns>A response value, in an async scenario will be T of Task of T</returns>
+        public static CallTargetReturn<TResponse> OnMethodEnd<TTarget, TResponse>(TTarget instance, TResponse response, Exception exception, CallTargetState state)
+            where TResponse : IElasticsearchResponse
+        {
+            var uri = response?.ApiCall?.Uri?.ToString();
+
+            if (uri != null)
+            {
+                var tags = (ElasticsearchTags)state.State;
+
+                if (tags != null)
+                {
+                    tags.Url = uri;
+                }
+            }
+
+            state.Scope.DisposeWithException(exception);
```

Comment:
Scope could be null if integration is disabled
```suggestion
            state.Scope?.DisposeWithException(exception);
```

Reply:
`DisposeWithException` has a null-check integrated: https://github.com/DataDog/dd-trace-dotnet/blob/a72b74fc8e67d8d8a6430628fe8643b3a693d2bc/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AutoInstrumentationExtensions.cs#L16

Reply:
Oh yeah. I feel like I commented for someone to remove the null check recently 🤦‍♂️